00new_packages.autobuild

```diff
@@ -1,4 +1,7 @@
 # IMPORTANT: new packages must be added at the top of this file
 # They must NEVER be added to other files
 
-
+cmake_package 'gui/rock_webapp'
+ruby_package 'tools/rest_api' do |pkg|
+    Autoproj.env_add_path 'ROCK_WEBAPP', File.join(pkg.prefix, 'share', 'rest_api')
```

Comment:
The variable is used to link "tools/rest_api" to the html part in "gui/webapp", as the html part hat to be provided by the rest_api on the same port as the api in order to be able to use the api with JS without the need to enable CORS (I hope you remember).

The variable is meant to hold several paths to html/js pages meant to access the rest api. It should be extenable for specific pages for single robots (like robot specific control UIs). So the variable holds a list of paths.

I think the point of discussion is about how to use the ROCK_PREFIX_PATH proposed by you.

IMHO it should be evaluated on varaible (environent) level and not on source level:

export ROCK_WEBAPP_PATH=${ROCK_PREFIX_PATH}/gui/webapp:${ROCK_PREFIX_PATH}/gui/ASIMO/webapp

This way custom (global) paths can be added as well as the prefix path can be used and changed.